### PR TITLE
Migrate ddc-client package to new storage architecture

### PR DESCRIPTION
**Pull request by skambalin**
Migrate `ddc-client` package to use new storage (MVP version) .

The PR also includes changes in other pacakges:
- file-storage: Transform input stream to fixed 64 kb chunk stream
- ddc - replace `bigint` with `number`

### CODE REVIEW

**Review by Argishtyfirst — APPROVED** [review]

File: packages/ddc-client/src/DdcClient.ts [hunk 1]
```diff
@@ -26,65 +16,34 @@ import {
...
-    ): Promise<DdcClient> {
-        encryptionSecretPhrase = encryptionSecretPhrase != null ? encryptionSecretPhrase : secretPhrase;
-        const clientCreateOptions = initDefaultClientOptions(options);
-
-        const scheme = isSchemeName(clientCreateOptions.scheme)
-            ? await Scheme.createScheme(clientCreateOptions.scheme, secretPhrase)
-            : clientCreateOptions.scheme;
+        readonly smartContract: SmartContract,
+        private scheme: Scheme,
+        private fileStorage: FileStorage,
```
Comment (upalinski) [inline packages/ddc-client/src/DdcClient.ts:30]: might be we should hide from out client initialisation of low-level APIs?

File: packages/ddc-client/src/DdcClient.ts [hunk 2]
```diff
@@ -26,65 +16,34 @@ import {
...
+        private router: Router,
+    ) {}
 
-        const smartContract = await SmartContract.buildAndConnect(secretPhrase, options.smartContract);
-        const caStorage = await ContentAddressableStorage.build(
-            {...clientCreateOptions, smartContract, scheme},
-            secretPhrase,
-        );
+    static async buildAndConnect(config: DdcClientConfig, secretPhrase: string) {
+        const scheme = await Scheme.createScheme('sr25519', secretPhrase);
```
Comment (upalinski) [inline packages/ddc-client/src/DdcClient.ts:35]: before the scheme (sr25519) was it options. why do we removed it and decided to use hard code?

File: packages/ddc-client/src/DdcUri.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export type DdcEntity = 'file' | 'dag-node';
+
+export type FileUri = DdcUri<'file'>;
+export type DagNodeUri = DdcUri<'dag-node'>;
+
+export class DdcUri<T extends DdcEntity = DdcEntity> {
+    constructor(readonly bucketId: number, readonly cid: string, readonly entity: T) {}
```
Comment (upalinski) [inline packages/ddc-client/src/DdcUri.ts:7]: actually, content type can be extracted from CID. no need to sepcify separate parameter. please, check this doc https://www.notion.so/cere/ADR-Protocol-interoperability-and-future-proofing-aaccc53547e84b5e93a9ca334d68e372

and out encoding component https://github.com/Cerebellum-Network/ddc-storage-node/blob/8322a64ece2fd11febe83fddcb7c337aeb05bede/pkg/cid/cid.go#L27C1-L27C1

File: packages/ddc/src/Router/Router.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+export type RouterNode = {
+    rpcHost: string;
+};
+
+export class Router {
+    constructor(private nodes: RouterNode[]) {}
+
+    async getNode(operation: RouterOperation) {
```
Comment (upalinski) [inline packages/ddc/src/Router/Router.ts:23]: looks like you don't use RouterOperation here. What is a purpose of this enum?

Comment (skambalin) [reply packages/ddc/src/Router/Router.ts:23]: For now it is a very temporary approach just to be able to integrate into real env without blockchain pallets. In general, based on the operation later will decide what node to choose (as it was done in our DAC v3 attempt) in future. But we can postpone the discussion, I think.

Comment (upalinski) [reply packages/ddc/src/Router/Router.ts:23]: not sure it make sense in our new approach  (v4 i guess ? :) ) okey, we can return to this discussion when we will implement real rotuing via pallet

File: packages/ddc-client/src/DdcUri.ts (hunk 3 above)
Comment (skambalin) [reply packages/ddc-client/src/DdcUri.ts:7]: It is also needed for build time type checking and type hinting (typescript).

Comment (skambalin) [reply packages/ddc-client/src/DdcUri.ts:7]: But it is nice to know in general, that we can detect this from cid

File: packages/ddc-client/src/DdcClient.ts (hunk 1 above)
Comment (skambalin) [reply packages/ddc-client/src/DdcClient.ts:30]: The constructor is protected, so I don't think we expose the API here. And also I am not sure that `FileStorage` is a low level API. The package can be used directly if the client don't need bucket/customer management and only wants to read/store files. At least it how it was in past and I think it is a nice option.

File: packages/ddc/src/Router/Router.ts (hunk 4 above)
Comment (skambalin) [reply packages/ddc/src/Router/Router.ts:23]: One example might be is read and store. When we want read data from CDN and store to Storage nodes. Router can decide what node to provide for the operation.

**Pull request merged**